services/app-web/src/gqlHelpers/apolloQueryWrapper.ts

```diff
@@ -0,0 +1,89 @@
+type QueryLoadingType = {
+    status: 'LOADING'
+}
+
+type QueryErrorType = {
+    status: 'ERROR'
+    error: Error
+}
+
+type QuerySuccessType<SuccessType, AdditionalVars = Record<string, unknown>> = {
+    status: 'SUCCESS'
+    data: SuccessType
+} & AdditionalVars
+
+type ApolloResultType<SuccessType, AdditionalVars = Record<string, unknown>> =
+    | QueryLoadingType
+    | QueryErrorType
+    | QuerySuccessType<SuccessType, AdditionalVars>
+
+// These are the parts of the useQuery result we want to wrap up, converting data | error | loading => result
+interface WrappableApolloResultsType {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    data: any
+    error?: Error
+    loading: boolean
+}
+
+type WrappedApolloResultType<
+    ResultType extends WrappableApolloResultsType,
+    AdditionalVars = Record<string, unknown>
+> = Omit<ResultType, 'data' | 'loading' | 'error'> & {
+    result: ApolloResultType<NonNullable<ResultType['data']>, AdditionalVars>
+}
+
+// Apollo's useQuery returns three independent variables to track loading, error, and data. In our usage
```

Comment:
This is such weird behavior from Apollo.  Love this wrapper.

Reply:
I guess it's not _weird_ exactly--other libraries do it, too, but I prefer the wrapped method.